cfn-resources/teams/cmd/resource/resource.go

```diff
@@ -0,0 +1,432 @@
+	}
+	return modelRole
+}
+func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup() // logger setup
+
+	_, _ = logger.Debugf("Update Team Request :%+v", currentModel)
+	// Validate required fields in the request
+	event, client, err := ValidateRequest(ReadRequiredFields, currentModel)
+	if err != nil {
+		return event, err
+	}
+	isExist := isExist(client, currentModel)
+	if !isExist {
+		_, _ = logger.Debugf("error getting Team information: %s", *currentModel.TeamId)
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          "Resource Not Found",
+			HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}, nil
+	}
+
+	// API call
+	team, res, err := client.Teams.Get(context.Background(), *currentModel.OrgId, *currentModel.TeamId)
+	if err != nil {
+		_, _ = logger.Debugf("error getting Team information: %s", err)
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          "Resource Not Found",
+			HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}, nil
+	}
+	_, _ = logger.Debugf("Read -Team information status (%d)", res.StatusCode)
+
+	// rename the team
+	if team.Name != *currentModel.Name {
+		_, _, err := client.Teams.Rename(context.Background(), *currentModel.OrgId, *currentModel.TeamId, *currentModel.Name)
+		if err != nil {
+			_, _ = logger.Debugf("error updating Team information: %v", err)
+		}
+	}
+
+	// add/remove user to/from teams
+	if currentModel.Usernames != nil {
+		// get the current team's users
+		users, _, err := client.Teams.GetTeamUsersAssigned(context.Background(), *currentModel.OrgId, *currentModel.TeamId)
+		if err != nil {
+			_, _ = logger.Debugf("Read -error (%v)", err)
+		}
+		usernames := currentModel.Usernames
+		var newUsers []string
+		for ind := range usernames {
+			currentUser, isExistingUser := isUserExist(users, usernames[ind])
+			_, _ = logger.Debugf("team -current user %v \n", currentUser)
+			if isExistingUser {
+				_, _ = logger.Debugf("team -remove user starts \n")
+				_, err := client.Teams.RemoveUserToTeam(context.Background(), *currentModel.OrgId, *currentModel.TeamId, currentUser.ID)
+				if err != nil {
+					_, _ = logger.Debugf("Remove Team -error (%v) \n", err)
+				}
+			} else {
+				_, _ = logger.Debugf("team -append user starts \n")
+				user, _, err := client.AtlasUsers.GetByName(context.Background(), usernames[ind])
+				if err != nil {
+					_, _ = logger.Debugf("Error reading user (%s)  with error (%v) \n", usernames[ind], err)
+				}
+				// if the user exists, we will store its ID
+				if user != nil {
+					newUsers = append(newUsers, user.ID)
+				}
+			}
+		}
+		_, _ = logger.Debugf("team -Add user starts \n")
+		_, _, err = client.Teams.AddUsersToTeam(context.Background(), *currentModel.OrgId, *currentModel.TeamId, newUsers)
+		if err != nil {
+			_, _ = logger.Debugf("team -Add user response status (%d)", res.StatusCode)
+		}
+	}
```

Comment:
this piece of code has a lot of nested if. Could you refactor it? moving some parts to specific functions would improve readability